source/MRMesh/MRMarchingCubes.cpp

```diff
@@ -529,24 +416,30 @@ Expected<TriMesh> volumeToMesh( const V& volume, const MarchingCubesParams& para
 
             SeparationPointSet set;
             bool atLeastOneOk = false;
-            const float baseValue = acc.get( baseLoc );
-
-            for ( int n = int( NeighborDir::X ); n < int( NeighborDir::Count ); ++n )
+            const float baseValue = cache ? cache->get( baseLoc ) : acc.get( baseLoc );
+            if ( !nanChecker( baseValue ) )
             {
-                bool ok = false;
-                Vector3f pos;
-                if ( cache )
-                    ok = findSeparationPoint( pos, volume, *cache, baseLoc, baseValue, NeighborDir( n ), params, std::forward<NaNChecker>( nanChecker ), std::forward<Positioner>( positioner ) );
-                else
-#ifndef MRMESH_NO_OPENVDB
-                if constexpr ( std::is_same_v<V, VdbVolume> )
-                    ok = findSeparationPoint( pos, volume, acc, baseLoc, baseValue, NeighborDir( n ), params, std::forward<Positioner>( positioner ) );
-                else
-#endif
-                    ok = findSeparationPointAcc( pos, acc, indexer, volume.voxelSize, baseLoc, baseValue, NeighborDir( n ), params, std::forward<NaNChecker>( nanChecker ), std::forward<Positioner>( positioner ) );
+                const auto baseCoords = zeroPoint + mult( volume.voxelSize, Vector3f( baseLoc.pos ) );
+                const bool baseLower = baseValue < params.iso;
 
-                if ( ok )
+                for ( int n = int( NeighborDir::X ); n < int( NeighborDir::Count ); ++n )
                 {
+                    auto nextLoc = baseLoc;
+                    nextLoc.pos[n] += 1;
+                    if ( nextLoc.pos[n] >= indexer.dims()[n] )
+                        continue;
+                    nextLoc.id = indexer.getExistingNeighbor( baseLoc.id, cPlusOutEdges[n] );
```

Comment:
Add a similar method for `VoxelLocation`?

Reply:
good idea, but I would prefer to do it in next PR